### PR TITLE
Add user email verification

### PR DESCRIPTION
**Pull request by ae2079**
#16

### CODE REVIEW

File: src/adapters/notifications/NotificationCenterAdapter.ts [hunk 1]
```diff
@@ -90,6 +90,29 @@ export class NotificationCenterAdapter implements NotificationAdapterInterface {
...
+    token: string;
+  }): Promise<void> {
+    const { email, user, token } = params;
+    try {
+      await callSendNotification({
+        eventName: NOTIFICATIONS_EVENT_NAMES.SEND_EMAIL_CONFIRMATION,
+        segment: {
+          payload: {
+            email,
+            verificationLink: `${dappUrl}/verification/${user.walletAddress}/${token}`,
```
Comment (aminlatifi) [inline src/adapters/notifications/NotificationCenterAdapter.ts:107]: This verification link format overlaps with project email verification.
```suggestion
            verificationLink: `${dappUrl}/verification/user/${user.walletAddress}/${token}`,
```

File: src/resolvers/userResolver.ts [hunk 2]
```diff
@@ -231,4 +235,115 @@ export class UserResolver {
...
+      }
+
+      const userToVerify = await findUserById(userId);
+
+      if (!userToVerify) {
+        throw new Error(i18n.__(translationErrorMessagesKeys.USER_NOT_FOUND));
+      }
+      if (userToVerify.id !== currentUserId) {
+        throw new Error(i18n.__(translationErrorMessagesKeys.UN_AUTHORIZED));
+      }
```
Comment (aminlatifi) [inline src/resolvers/userResolver.ts:257]:
```suggestion
```

Comment (aminlatifi) [reply src/resolvers/userResolver.ts:257]: If the above is accepted

File: src/resolvers/userResolver.ts [hunk 3]
```diff
@@ -231,4 +235,115 @@ export class UserResolver {
...
+
+      const email = userToVerify.email;
+      if (!email) {
+        throw new Error(
+          i18n.__(translationErrorMessagesKeys.NO_EMAIL_PROVIDED),
+        );
+      }
+      if (userToVerify.emailConfirmed) {
```
Comment (aminlatifi) [inline src/resolvers/userResolver.ts:265]: What if the user wants to change the email?!
Project Verification form has considered this case

**Review by aminlatifi — CHANGES_REQUESTED** [review]
Good job @ae2079 
Just few amendments

File: src/resolvers/userResolver.ts (hunk 3 above)
Comment (ae2079) [reply src/resolvers/userResolver.ts:262]: I should make emailVerification fields empty in the change email flow

**Review by aminlatifi — APPROVED** [review]

**Comment by aminlatifi** [comment]
Thanks @ae2079

**Pull request merged**